posthog/hogql/transforms/lazy_tables.py

```diff
@@ -252,3 +394,12 @@ def visit_select_query(self, node: ast.SelectQuery):
                 field_or_property.joined_subquery = table_type
 
         self.stack_of_fields.pop()
+
+        # When joining a lazy table to another lazy table, the joined table doesn't get resolved
+        # Doing another pass solves this for us
+        if self.lazy_finder_counter < 20:
+            lazy_finder = LazyFinder()
+            lazy_finder.visit(node)
+            if lazy_finder.found_lazy:
+                self.lazy_finder_counter = self.lazy_finder_counter + 1
+                self.visit_select_query(node)
```

Comment:
This feels like code nobody will dare touch for years 😅. I wonder if there's a smarter way to just flag tables that should be revisited, instead of doing it for everything?